src/proto/streams/store.rs

```diff
@@ -12,6 +12,11 @@ use std::ops;
 pub(super) struct Store {
     slab: slab::Slab<(StoreId, Stream)>,
     ids: OrderMap<StreamId, (usize, StoreId)>,
+    /// Keeps a list of StreamIds that we have reset locally, to allow ignoring
+    /// frames on that stream "for some time".
+    // We could store an `Instant` here perhaps, and upon lookup, if the
+    // elapsed time has been too long, pop it from this Map and return false.
+    reset_streams: OrderMap<StreamId, ()>,
```

Comment:
Ideally we wouldn't introduce a second map. Would it be possible to track the locally reset state as part of the stream state and have a single lookup path?

It also looks like introducing a second map doubles the number of hash lookups for all received data frames (from below).

Reply:
So, we *could* keep the streams around afterwards, but a `Stream` is rather big in bytes, and we only care about a) the ID, b) that it was reset locally, c) how long ago. Additionally, locally resetting shouldn't be that uncommon, but hopefully receiving frames on a reset stream is less common, and so having a second map is the better situation...

Reply:
@seanmonstar As of now, I wouldn't worry about the byte size of `Stream`. We will be shrinking it significantly.